nexus/db-queries/src/db/datastore/instance.rs

```diff
@@ -2959,6 +2966,7 @@ mod tests {
         assert_eq!(instance.runtime().dst_propolis_id, Some(vmm3.id));
 
         // Clean up.
+        datastore.terminate().await;
         db.cleanup().await.unwrap();
         logctx.cleanup_successful();
```

Comment:
How hard would it be to abstract this out into a cleanup function? Seems a little excessive to have it add it everywhere.

Making this an abstract function would also help with things like ensuring the right order. (For example, can `db.cleanup().await` and `datastore.terminate().await` be interchanged? Is it valid to use `tokio::join!` across both?

Reply:
I'm taking this on as a follow-up PR, to refactor the testing utilities and make this usable outside nexus-db-queries